### PR TITLE
create-service-stubs

### PR DESCRIPTION
**Pull request by Rizziepit**

### CODE REVIEW

**Comment by Rizziepit** [comment]
@miltontony ready for review. This implements GET and POST for user data, and it authenticates the requesting app via basic auth.

PS. The important files are `models.py`, `user_api.py`, `test_auth.py` and `test_user_api.py`

File: unicore/hub/service/tests/test_auth.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+from unicore.hub.service.tests import DBTestCase
+
+
+class AuthTestCase(DBTestCase):
+
+    def test_app_authentication(self):
+
+        def requestUser(credentials=None):
```
Comment (miltontony) [inline unicore/hub/service/tests/test_auth.py:8]: nit-pick: This should be `requestApp`

Comment (Rizziepit) [inline unicore/hub/service/tests/test_auth.py:8]: It's requesting user data though?

File: unicore/hub/service/user_api.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    key = str(app.id)
+    return app_data.get(key, {})
+
+
+@user_app_data.post()
+def save_user(request):
+    app = get_authenticated_app_object(request)
+    user = get_user_object(request)
+
+    # NB: always convert key to string because JSONType field
```
Comment (miltontony) [inline unicore/hub/service/user_api.py:49]: more reason why `app.id` should be a slug

File: unicore/hub/service/tests/test_auth.py (hunk 1 above)
Comment (miltontony) [inline unicore/hub/service/tests/test_auth.py:8]: aah..it looked like it was getting the app. Ignore me ;)

**Comment by miltontony** [comment]
:+1:

File: unicore/hub/service/user_api.py (hunk 2 above)
Comment (Rizziepit) [inline unicore/hub/service/user_api.py:49]: Working on it right now

**Pull request merged**